### PR TITLE
Fix the 404 exception when an issue isn't found

### PR DESCRIPTION
**Pull request by mgust**
- More debugging
- Built the code as well

### CODE REVIEW

File: src/jira.ts [hunk 1]
```diff
@@ -174,11 +170,7 @@ Valid sample branch names:
   };
 
   /** Check if jira project validation is enabled then compare the issue project with the allowed statuses. */
-  static isProjectValid = (
-    shouldValidate: boolean,
-    allowedProjects: string[],
-    details: JIRADetails
-  ): boolean => {
+  static isProjectValid = (shouldValidate: boolean, allowedProjects: string[], details: JIRADetails): boolean => {
```
Comment (mohamed-akef) [inline src/jira.ts:173]: Please check the project coding standards and flow it if possible

File: src/main.ts [hunk 2]
```diff
@@ -147,19 +147,20 @@ async function run(): Promise<void> {
...
+      return exit('JIRA issue id is missing in your branch or PR title.');
     }
 
     // use the last match (end of the branch name)
     const issueKey = issueKeys[issueKeys.length - 1];
     console.log(`JIRA key -> ${issueKey}`);
 
-    const details: JIRADetails = await jira.getTicketDetails(issueKey);
-    if (details.key) {
+    try {
```
Comment (mohamed-akef) [inline src/main.ts:161]: Why are we allowing getTicketDetails to throw an exception, We can make it return null if there is no data and keep the if condition as is

**Review by mohamed-akef — APPROVED** [review]
There is no blocker for merging it; all comments are improvements

File: src/main.ts (hunk 2 above)
Comment (mgust) [reply src/main.ts:161]: I tried this, but that meant amending the JIRADetails interface to be nullable, which made the assignment break in hilarious ways, et.c, et.c
So yes, this is an issue of skipping abstraction layers, but I must admit my TypeScript isn't good enough to do it in the correct way. Very happy for code suggestions of how to do it correctly. :)

File: src/jira.ts (hunk 1 above)
Comment (mgust) [reply src/jira.ts:173]: Could you clarify what you think is wrong here? It is pretty much a copy-pasted definition from one of the previous functions, and then the npm prettify is sorting out the indentation et.c

Comment (mohamed-akef) [reply src/jira.ts:173]: Yes, but with a different style.
you changed quoteProps style; I really didn't know what of them was matching the prettierrc config but you need to make sure you are confined prettier and eslint correctly
https://github.com/reliancehealthinc/action-jira-linter/blob/main/.prettierrc.js
https://github.com/reliancehealthinc/action-jira-linter/blob/main/.eslintrc.js

File: src/main.ts (hunk 2 above)
Comment (mohamed-akef) [reply src/main.ts:161]: Sure you are welcome; you will just need to allow undefined as a type.
  const details: JIRADetails | undefined = await jira.getTicketDetails(issueKey);

I know TS not look like other languages but by time you will like it :D

**Pull request merged**